### DISCUSSION TITLE
PWA-LAST-Stack: All the latest Symfony stuff in one repo

### DISCUSSION THREAD

**Discussion by tacman** [Spomky-Labs/pwa-bundle#115] [Show and tell]
I took the course code from https://symfonycasts.com/screencast/last-stack, which shows off the latest in Symfony (LAST = Live Components, AssetMapper, Stimulus, Turbo), and add pwa-bundle.

See it in action here: https://pwa-last-stack.survos.com/ 
Get the source here: https://github.com/survos-sites/pwa-last-stack

Play with it (and checkout the cool debug toolbar) 

```bash
git clone git@github.com:survos-sites/pwa-last-stack.git && cd pwa-last-stack
composer install
echo "DATABASE_URL=sqlite:///%kernel.project_dir%/var/data.db" > .env.local
bin/console d:schema:update --force --complete
bin/console doctrine:fixtures:load -n
bin/console tailwind:build
symfony server:start -d
symfony open:local
```

Things I need help with:

* The images for the logo in the debug toolbar (PWA) aren't right, and I'm not sure how to correct that.
* The screenshots sometimes work -- a caching issue?
* If the app is installed, there should be a button to launch it if you're on the web page.
* I'm not sure it's caching as expected.  My idea is that the planets (since there's a limited number) and the index pages should be CacheThenNetwork, the planets pages should be pre-loaded so you can go offline.  
* I temporarily disabled Turbo because it has it's own caching system.  When I put it back, I need to add more routes that are partial pages.

Feedback welcome and appreciated!

**Comment by Spomky** [comment]
That's really great.
I modified a few things and will dig into the other problems you mentioned.
I encountered the same issues with Turbo. Not sure how to deal with it, especially the stream feature.

**Comment by tacman** [comment]
I added an about page with a QR code, so I could test in on a mobile device and install it.  I'm practically delirious clicking "Install as PWA" switching to Airplane mode, going to my apps and opening it and navigating around.  This is the stuff of geek fantasy.  

![image](https://github.com/Spomky-Labs/pwa-bundle/assets/619585/e3a6302b-33a4-4e6c-9852-3984b3eb5d15)

I think this is a good repo for real-world testing, but it's not ready for prime time.  I'm misunderstanding something with caching, the offline fallback of the home page happens on the planets that should already be preCached.  

Also, it's not mobile-friendly, and I'm struggling with tailwind.  But I'll tweak it somehow.




https://pwa-last-stack.survos.com/about
